endpoint.go

```diff
@@ -222,6 +228,15 @@ func (ep *endpoint) isAnonymous() bool {
 	return ep.anonymous
 }
 
+func (ep *endpoint) needDNSServer() bool {
+	if ep.getNetwork().Type() == "null" || ep.getNetwork().Type() == "host" {
+		return false
+	}
```

Comment:
Can't we have the above endpoint option set for null and host network as well ?
Why do we need this explicit check here ?